Sources/Classes/in-app-purchase/MiniAppIAPModule.swift

```diff
@@ -0,0 +1,76 @@
+import Foundation
+import StoreKit
+
+public class MiniAppIAPModule: NSObject, SKProductsRequestDelegate, SKPaymentTransactionObserver {
+
+    public typealias ProductResponseCompletionHandler = ((Result<MAProductResponse, MAProductResponseError>)) -> Void
+    var productResponseHandlerObj: ProductResponseCompletionHandler?
+    var paymentProductObj: SKProduct?
+    private var models = [SKProduct]()
+    var productId: String = ""
+
+    public func initiateProductPayment(productId: String, completionHandler: @escaping ProductResponseCompletionHandler) {
+        self.productResponseHandlerObj = completionHandler
+        self.productId = productId
+        let productRequest = SKProductsRequest(productIdentifiers: Set(["com.temporary.id"]))
```

Comment:
Just a question: the product identifiers are hard coded here is this intended for the current state?

Reply:
Sorry, this was meant to be cleaned up while working on testcases. Thanks for noticing. Removed it now